### PR TITLE
Fix line offset when using single-line ~H variants

### LINKED ISSUE #240: Fix line offset when using single-line ~H variants
Before Elixir `v1.10` we were not able to report the error line correctly for single-line `~H` because there was no way to retrieve the information about the delimiter so we had to add a `+ 1` to the line offset because using `~H"""` is more common than using the other single-line variants like `~H"..."` and `~H(...)`. See:

https://github.com/msaraiva/surface/blob/2286f8b6130679f60fd5ece18251d3f84a1dd830/lib/surface.ex#L106-L109

Since `v1.10`, the delimiter is available so we're now able to fix this. Here's the difference:

### ~H""" in Elixir `v1.9.1`

```elixir
{:sigil_H, [], [{:<<>>, [], ["text\n"]}, []]} 
```

### ~H() in Elixir `v1.9.1`

```elixir
{:sigil_H, [], [{:<<>>, [], ["text"]}, []]}
```

### ~H""" in Elixir `v1.11.1`

```elixir
{:sigil_H, [delimiter: "\"\"\""], [{:<<>>, [indentation: 2], ["text\n"]}, []]}
```

### ~H() in Elixir `v1.11.1`

```elixir
{:sigil_H, [delimiter: "("], [{:<<>>, [], ["text"]}, []]}
```

### PR DESCRIPTION
**Pull request by Malian**
Fixes #240

### CODE REVIEW

**Comment by Malian** [comment]
I have no idea why it fails with version 1.10 as the fix has been merged the [15 jan 2020](
https://github.com/elixir-lang/elixir/pull/9738) before the release of the 1.10.

I will investigate more this afternoon.

File: lib/surface.ex [hunk 1]
```diff
@@ -123,6 +121,18 @@ defmodule Surface do
...
   end
 
+  defp compute_line_offset(meta) do
+    # Since `v1.10` this will create accurate line numbers for heredoc usages
+    # of the `~H` sigil. For version below `v1.10` single-line ~H will return
+    # an incorrect line number because there was no way to retrieve the information
+    # about the delimiter. See https://github.com/msaraiva/surface/issues/240
+    cond do
+      Keyword.has_key?(meta, :indentation) -> 1
+      not Version.match?(System.version(), "~> 1.10") -> 1
```
Comment (Malian) [inline lib/surface.ex:131]: Should we use the condition `~> 1.10` or `>= 10.0.0`? The last one will work with Elixir 2.0.

File: lib/surface.ex [hunk 2]
```diff
@@ -123,6 +121,18 @@ defmodule Surface do
...
   end
 
+  defp compute_line_offset(meta) do
+    # Since `v1.11` this will create accurate line numbers for heredoc usages
+    # of the `~H` sigil. For version below `v1.11` single-line ~H will return
+    # an incorrect line number because there was no way to retrieve the information
+    # about the delimiter. See https://github.com/msaraiva/surface/issues/240
+    cond do
+      Keyword.has_key?(meta, :indentation) -> 1
+      not Version.match?(System.version(), "~> 1.11") -> 1
```
Comment (Malian) [inline lib/surface.ex:131]: Should we use the condition `~> 1.11` or `>= 1.11.0`? The last one will work with Elixir 2.0.

**Comment by Malian** [comment]
> I have no idea why it fails with version 1.10 as the fix has been merged the [15 jan 2020](https://github.com/elixir-lang/elixir/pull/9738) before the release of the 1.10.
> 
> I will investigate more this afternoon.

The ` :indentation` key is only available since elixir `v1.11`

@msaraiva Ready to review

File: test/compiler_test.exs [hunk 3]
```diff
@@ -802,11 +802,14 @@ defmodule Surface.CompilerSyncTest do
...
+    line_offset = if Version.match?(System.version(), "~> 1.11"), do: 1, else: 0
+
     output =
       capture_io(:standard_error, fn ->
-        # Setting line to 0 here because we aren't using heredoc for the above code and so the lines would
-        # be off
-        {{:module, module, _, _}, _} = Code.eval_string(component_code, [], %{__ENV__ | line: 0})
+        {{:module, module, _, _}, _} =
+          Code.eval_string(component_code, [], %{__ENV__ | line: line_offset})
+
```
Comment (msaraiva) [inline test/compiler_test.exs:812]: ❤️

**Review by msaraiva — APPROVED** [review]

**Comment by msaraiva** [comment]
@Malian thanks! ❤️

**Pull request merged**